npm/vite-dev-server/src/resolveConfig.ts

```diff
@@ -4,6 +4,7 @@
  * You can find it here https://github.com/vitest-dev/vitest/blob/main/packages/vitest/src/node/create.ts
  */
 import debugFn from 'debug'
+import { searchForWorkspaceRoot } from 'vite'
```

Comment:
Looks like this is causing CI to fail - I think the issue here is we don't bundle `vite` - that's from userland. So during the build step for Cypress, this doesn't actually exist.

We have something similar landing https://github.com/cypress-io/cypress/pull/26066/files#diff-5617aebe2ce067468c76a778abff48d659d1a28b47e93b2d0fff69f27cb2119eR61

We can either:

1. just copy paste from vite: https://github.com/vitejs/vite/blob/6b40f03574cd71a17cbe564bc63adebb156ff06e/packages/vite/src/node/server/searchRoot.ts#L59
2. create a npm package that does this? might be useful. Then ship it with the Cypress app
3. source the user's vite config and load via dynamic import, eg

```ts
export const createViteDevServerConfig = async (config: ViteDevServerConfig, vite: Vite): Promise<InlineConfig> => {
  const vite = await import('vite')
  const { findWorkspaceRoot } = vite
```

I think (2) or (3) is good. Thoughts?

Reply:
@lmiller1990 what is the second argument passed to `createViteDevServerConfig`? Is it Vite instance itself?

I'm pretty lost here on how cypress works so I'm wondering if you can't use the `findWorkspaceRoot` from `vite` argument

Reply:
Huh, I think you are correct - nice catch, it looks like we already solved this at some point (been a while, I forgot how this works exactly).

We do have access to the user's vite, see here: https://github.com/cypress-io/cypress/blob/9c98b5e08583764ef228d9b8a40e5f28e7f41ade/npm/vite-dev-server/src/resolveConfig.ts#L54

Go for it, want to update this and do `vite.findWorkspaceRoot` using the Vite instance we receive? That should work.